src/main/java/org/umcn/me/util/Reference.java

```diff
@@ -0,0 +1,128 @@
+package org.umcn.me.util;
+
+import net.sf.picard.reference.*;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+
+public class Reference {
+    private static ReferenceSequenceFile reference = null;
+    private static FastaSequenceIndex index = null;
+
+    public static void loadReference(String refPath) throws IOException {
+
+        //Check if the reference exists
+        File refFile = new File(refPath);
+        if(!refFile.exists())
+            throw new FileNotFoundException("The reference genome was not found");
+
+        //If index doesn't exist at the same location as the reference, create it
+        String indexPath = refPath.replaceFirst("\\.[a-z]+$",".fai");
+
+        File indexFile = new File(indexPath);
+        if(!indexFile.exists()){
+            try {
+                createIndex(refPath, indexPath);
+            } catch (IOException e) {
+                throw new IOException("Could not create reference index file: " + e.toString());
+            }
+        }
+
+        //Load the index
+        FastaSequenceIndex index = new FastaSequenceIndex(indexFile);
+
+        //Load the reference
+        reference = new IndexedFastaSequenceFile(new File(refPath), index);
+    }
+
+    static public void createIndex(String refPath, String indexPath) throws IOException {
+
+        try (BufferedReader refRead = new BufferedReader(new InputStreamReader(new FileInputStream(refPath))); BufferedWriter indexWrite = new BufferedWriter(new FileWriter(indexPath))) {
```

Comment:
lots of stacking here.
general code style would be not make the line so long or embed constructors so deeply